Framework/Algorithms/src/Rebunch.cpp

```diff
@@ -82,24 +83,20 @@ void Rebunch::exec() {
   PARALLEL_FOR2(inputW, outputW)
   for (int hist = 0; hist < histnumber; hist++) {
     PARALLEL_START_INTERUPT_REGION
-
-    // get const references to input Workspace arrays (no copying)
-    const MantidVec &XValues = inputW->readX(hist);
-    const MantidVec &YValues = inputW->readY(hist);
-    const MantidVec &YErrors = inputW->readE(hist);
-
-    // get references to output workspace data (no copying)
-    MantidVec &XValues_new = outputW->dataX(hist);
-    MantidVec &YValues_new = outputW->dataY(hist);
-    MantidVec &YErrors_new = outputW->dataE(hist);
-
     // output data arrays are implicitly filled by function
     if (point) {
-      rebunch_point(XValues, YValues, YErrors, XValues_new, YValues_new,
-                    YErrors_new, n_bunch);
+      rebunch_point(inputW->x(hist), inputW->y(hist), inputW->e(hist),
+                    outputW->mutableX(hist), outputW->mutableY(hist),
+                    outputW->mutableE(hist), n_bunch);
```

Comment:
Since a `Histogram` knows if it is `point` or `dist` there also is no need for checking this here, that can be done internally in the `rebunch` method.